factory-reset-lvm

```diff
@@ -93,23 +93,35 @@ fi
 
 # Find the biggest versioned Sailfish folder and use that
 SAILFISH_FIMAGE=$(ls -d $FIMAGE_MOUNT/Sailfish* | tail -1)
+WORKDIR=$(pwd)
 
 if test -z $SAILFISH_FIMAGE; then
 	echo "$NAME: Error: Could not find a recovery image folder!" > /dev/kmsg
 	exit 1
 fi
 
-if ! test -f $SAILFISH_FIMAGE/sailfish_root.squashfs.img; then
+# Check that the factory images are ok to use.
+cd $SAILFISH_FIMAGE
+if test -f root.img.lzo; then
+	if ! md5sum -c root.img.lzo.md5; then
+		echo "$NAME: Error: root recovery image corrupted!" > /dev/kmsg
+		exit 1
+	fi
+else
 	echo "$NAME: Error: cannot find sailfish root recovery image!" > /dev/kmsg
 	exit 1
 fi
 
-if ! test -f $SAILFISH_FIMAGE/sailfish_home.squashfs.img; then
+if test -f home.img.lzo; then
+	if ! md5sum -c home.img.lzo.md5; then
```

Comment:
this could be:
```
if ! test -f home.img.lzo || ! md5sum -c home.img.lzo.md5; then
```
right? And same above. Also is it intended to print md5sum output if so shoul dit go to > /dev/kmsg?